conda/cli/install.py

```diff
@@ -54,14 +54,17 @@ def check_prefix(prefix, json=False):
                           "Spaces in paths can sometimes be problematic." % prefix)
 
 
-def clone(src_arg, dst_prefix, json=False, quiet=False, index_args=None):
+def clone(src_arg, dst_prefix, json=False, quiet=False, index_args=None, use_context=True):
     if os.sep in src_arg:
         src_prefix = abspath(src_arg)
         if not isdir(src_prefix):
             raise DirectoryNotFoundError(src_arg)
     else:
-        assert context._argparse_args.clone is not None
-        src_prefix = locate_prefix_by_name(context._argparse_args.clone)
+        if use_context:
+            assert context._argparse_args.clone is not None
+            src_prefix = locate_prefix_by_name(context._argparse_args.clone)
+        else:
+            src_prefix = locate_prefix_by_name(src_arg)
```

Comment:
```python
elif use_context:
    assert context._argparse_args.clone is not None
    src_prefix = locate_prefix_by_name(context._argparse_args.clone)
else:
    src_prefix = locate_prefix_by_name(src_arg)
```

Reply:
Wait, isn't `context._argparse_args.clone == src_arg`?

The `clone` method is only called in one place previously:
https://github.com/conda/conda/blob/ae389d58d67dd8e29952e669c34f1495d034185a/conda/cli/install.py#L217

And `context._argparse_args` _is_ the parsed args as set here:
https://github.com/conda/conda/blob/ae389d58d67dd8e29952e669c34f1495d034185a/conda/cli/main.py#L78

So from what I'm reading we can simply flatten all of this and remove the `use_context` arg:
```
else:
    src_prefix = locate_prefix_by_name(src_arg)
```

Reply:
Thanks for this. Yeah, I had made these changes before learning more about the context object itself. I will implement what you suggest.